kernel/process/process.rs

```diff
@@ -351,6 +356,38 @@ impl Process {
         self.signals.lock().is_pending()
     }
 
+    /// Sets signal mask
+    pub fn rt_sigprocmask(
+        &self,
+        how: usize,
+        set: Option<UserVAddr>,
+        oldset: Option<UserVAddr>,
+        _length: usize,
+    ) -> Result<isize> {
+        let mut sigset = self.sigset.lock();
+
+        if let Some(old) = oldset {
+            if let Err(_) = old.write_bytes(sigset.as_slice()) {
+                return Err(Error::new(Errno::EFAULT));
+            }
+        }
+
+        if let Some(new) = set {
+            if let Ok(new_set) = new.read::<[u8; 128]>() {
```

Comment:
Nit: I guess you can use `?` here.

```suggestion
            let new_set = new.read::<[u8; 128]>()?;
```

Reply:
Done